python/private/python.bzl

```diff
@@ -318,27 +357,80 @@ def _python_impl(module_ctx):
                 set_python_version_constraint = is_last,
             ))
             if _is_compatible_with_host(module_ctx, platform_info):
-                host_platforms[platform_name] = platform_info
+                host_compat_entry = struct(
+                    full_python_version = full_python_version,
+                    platform = platform_info,
+                    platform_name = platform_name,
+                    impl_repo_name = repo_name,
+                )
+                host_platforms[platform_name] = host_compat_entry
+                all_host_compatible_impls.setdefault(full_python_version, []).append(
+                    host_compat_entry,
+                )
+                all_host_compatible_impls.setdefault(
+                    full_python_version.rpartition(".")[0],
```

Comment:
nit, using the `version.bzl` to parse the `full_python_version` and then using the `major.minor` from that could be more robust.

But it's fine for this PR as maybe my preference to parse the version here is non warranted.

Reply:
Done, though I had to `v.release[0:2]` since there's no .major or .minor attributes of the parsed object.